### PR TITLE
chore(cargo-psp): update deps + use clap::Parser

### PR DESCRIPTION
**Pull request by SK83RJOSH**
Decided to fix up warnings in cargo-psp, as well as update all it's dependencies. As a bonus removed a bunch of clap boilerplate using it's `derive` feature. 🙂

### CODE REVIEW

File: cargo-psp/src/bin/mksfo.rs [hunk 1]
```diff
@@ -55,72 +54,75 @@ impl SfoEntry {
...
-            Arg::with_name("bare")
-                .long("bare")
-                .help("Do not set any default values. Ignores the <title> value if set."),
-        )
-        .arg(
-            Arg::with_name("dword")
-                .short("d")
-                .long("dword")
-                .help("key=VALUE Add a new DWORD value")
-                .multiple(true)
```
Comment (sajattack) [inline cargo-psp/src/bin/mksfo.rs:85]: this multiple flag seems to have been removed

**Review by sajattack — COMMENTED** [review]
In general I'm concerned that we may accidentally lose functionality by switching to the derive feature of clap, but if you can prove otherwise, the rest looks good and I'm happy to accept this.

File: cargo-psp/src/bin/mksfo.rs (hunk 1 above)
Comment (SK83RJOSH) [reply cargo-psp/src/bin/mksfo.rs:85]: I believe multiple is automatically implied by a field being `Vec` when using Derive - and I of course tested with a fresh run of `cargo psp` while fixing up `main.rs`. 🙂

**Comment by SK83RJOSH** [comment]
> In general I'm concerned that we may accidentally lose functionality by switching to the derive feature of clap, but if you can prove otherwise, the rest looks good and I'm happy to accept this.

Anything in particular you'd like me to test? `mksfo` was the most complex case, and appears to be working just fine from my testing. The rest were relatively straight-forward, with the exception of `pack-pbp`, since it used `NULL` instead of optional params. I special cased that one.

**Pull request merged**